src/core/coap/coap.cpp

```diff
@@ -786,17 +753,12 @@ otError ResponsesQueue::GetMatchedResponseCopy(const Header &          aHeader,
         }
 
         // Check Message Id
-        if (header.FromMessage(*message, sizeof(EnqueuedResponseHeader)) != OT_ERROR_NONE)
-        {
-            continue;
-        }
-
-        if (header.GetMessageId() != aHeader.GetMessageId())
+        if (message->GetMessageId() != aRequest.GetMessageId())
         {
             continue;
         }
 
-        *aResponse = message->Clone();
+        *aResponse = static_cast<Message *>(message->Clone());
```

Comment:
Is this `static_cast` necessary with the introduction of `ot::Coap::Message::Clone()` methods?